### ISSUE TITLE
Iterator zombie won't die :)

### ISSUE THREAD

**Issue by cefn** [tino/pyFirmata#13]
The Iterator thread doesn't die gracefully on KeyboardInterrupt or SystemExit, or when the Main thread completes.

Additionally, the function _Thread__stop() which does indeed kill the Iterator doesn't appear to be broken out to a nice function call, like Iterator.stop() as seems to be suggested within the Mockup implementation.

I propose setting the Iterator thread's isDaemon() flag to True in the Iterator constructor

``` python
        self.setDaemon(True)
```

If the developer chooses to call Iterator run() in the main thread (preventing the python interpreter from returning) they may need code like the following so it can be killed when foregrounded...

``` python
            except(KeyboardInterrupt, SystemExit), e:
                sys.exit()
```

...and a simple extra method to map the stop function to Thread's stopping function could be nice...

``` python
def stop(self):
    self._Thread__stop()
```

**Comment by NeoPolus** [comment]
Hi, this bug is due to Board.exit() never getting called automatically from Board.__del__() as long as there is a Iterator thread running.
This happens because the Iterator thread keeps a -circular- reference to the Board! So if you try to stop Python (keyboard interruption or any exception...) the main thread dies but Python does not call Board.__del__() yet and instead keeps waiting for the Iterator thread to stop... which never happens.
So, unless you manually call Board.exit() under any situation ("try:... finally: board.exit()") is highly probable that your Iterator will become a zombie.

I created a new pull request (https://github.com/tino/pyFirmata/pull/45) with some commits that fix this problem by implementing the support for the Python "with" statement, so you can do something like this:

>> from pyfirmata import Arduino, util
>> board = Arduino('/dev/tty...')
>> with board:
>>     board.analog[0].enable_reporting()
>>     board.analog[0].read()

and it will take care of creating the Iterator threading at the beginning of the with block and stopping it (calling board.exit()) at the end of the block even when an exception happens. So it would be a shorter (and cooler ;) ) equivalent to:

>> from pyfirmata import Arduino, util
>> board = Arduino('/dev/tty...')
>> it = util.Iterator(board)
>> it.start()
>> try:
>>     board.analog[0].enable_reporting()
>>     board.analog[0].read()
>> finally:
>>     board.exit()

**Comment by tino** [comment]
Fix with #44
